### ISSUE TITLE
PDM score of transfuser baseline

### ISSUE THREAD

**Issue by RexGPT** [autonomousvision/navsim#35]
Hi @DanielDauner , thank you for your excellent work! I've been running the Transfuser baseline according to the documentation, but I am encountering some discrepancies. Despite multiple runs, my results show the Transfuser baseline consistently fluctuating around 80.0 PDMS, which is lower than the reported 84.0 PDMS in the [NAVSIM paper](https://arxiv.org/pdf/2406.15349) but higher than the 78.0 PDMS reported in the [Hydra-MDP paper](https://arxiv.org/pdf/2406.06978). Could you provide any insights into this variation?

| Method | NC| DAC|TTC| Comf.| EP|PDMS| 
| :---: | :---: | :---: | :---: |  :---: | :---:|:---:|
| Transfuser in NAVSIM paper| 97.7|92.8| 92.8| 100|79.2| 84.0| 
| Transfuser in Hydra-MDP paper| 96.5|87.9| 90.2| 100|73.9| 78.0| 
|Mine reproducement|97.3|89.0|92.1|100|75.4|80.0

**Comment by DanielDauner** [comment]
Hi @RexGPT 

It's unfortunate that you can't reproduce the numbers. May I ask which data split you train on, what batch size you use, and for how many epochs you train?

We trained the TransFuser model on `navtrain` with a batch size of 64 (single GPU) and for 100 epochs. The settings had an impact on the performance. We adapted these parameters as default in NAVSIM v1.1. We also uploaded the checkpoints for TransFuser on [https://huggingface.co/autonomousvision/navsim_baselines](https://huggingface.co/autonomousvision/navsim_baselines).
 
I hope that helps in finding the issue!

Best,
Daniel

**Comment by RexGPT** [comment]
@DanielDauner , Thanks for your prompt reply!
> We trained the TransFuser model on navtrain with a batch size of 64 (single GPU) and for 100 epochs.

Did this mean that you only use a single GPU? I use 8 gpus and train the split on `trainval`, this is my command, which is exactly the same as the doc:

```bash
python $NAVSIM_DEVKIT_ROOT/navsim/planning/script/run_training.py \
        agent=transfuser_agent \
        experiment_name=training_transfuser_agent  \
        train_test_split=navtrain  \
        split=trainval   \
        trainer.params.max_epochs=100 \
        cache_path="${NAVSIM_EXP_ROOT}/training_cache/" \
        use_cache_without_dataset=True  \
        force_cache_computation=False 
```

**Comment by RexGPT** [comment]
Mine batch size is by default, and I use 8 GPUs:
```yaml
dataloader:
  params:
    batch_size: 64  # number of samples per batch
    num_workers: 4  # number of workers for data loading
    pin_memory: true  # pin memory for faster GPU transfer
    prefetch_factor: 2  # number of samples loaded in advance by each worker
```

**Comment by DanielDauner** [comment]
Hi @RexGPT 

Please note that the batch size is defined "per-GPU" (as NAVSIM uses PyTorch Lightning for training). If you use 8 GPUs, you have an effective batch size of 512, resulting in fewer parameter updates over 100 training epochs. 

To change the number of GPUs, you can add a `devices` argument to the `trainer` params of [here](https://github.com/autonomousvision/navsim/blob/main/navsim/planning/script/config/training/default_training.yaml#L31) (see also [docs](https://lightning.ai/docs/pytorch/2.2.1/common/trainer.html#devices)). Or define the visible cuda device in the bash script, e.g. 
```bash 
CUDA_VISIBLE_DEVICES=0  $NAVSIM_DEVKIT_ROOT/navsim/planning/script/run_training.py \
...
```

You can also experiment with training for more epochs or adapting the learning rate. You generally need to re-tune some parameters when increasing the batch size (or distributing training over multiple GPUs).

Best,
Daniel

**Comment by RexGPT** [comment]
Hi @DanielDauner , thanks for the clarification! Just to confirm, in your implementation, did you use only 1 GPU with a total batch size of 64? Did I understand that correctly?

**Comment by DanielDauner** [comment]
Yes, we only used 1 GPU with a batch size of 64. That's correct!

**Comment by RexGPT** [comment]
Many thanks! Your response is very helpful!

**Comment by mo704507064** [comment]
> Many thanks! Your response is very helpful!

We encounter the same problem，we use the uploaded checkpoints for TransFuser in Huggingface，evaluation on the test dataset. The result is about 80. [](url) 

![Image](https://github.com/user-attachments/assets/06da8634-b23a-4684-b59b-73c35aa51b72)
